packages/astro/src/cli/install-package.ts

```diff
@@ -0,0 +1,118 @@
+import boxen from 'boxen';
+import { execa } from 'execa';
+import { bold, cyan, dim, magenta } from 'kleur/colors';
+import ora from 'ora';
+import prompts from 'prompts';
+import whichPm from 'which-pm';
+import { debug, info, type LogOptions } from '../core/logger/core.js';
+
+type GetPackageOptions = {
+	skipAsk?: boolean;
+	cwd?: string;
+};
+
+export async function getPackage<T>(
+	packageName: string,
+	logging: LogOptions,
+	options: GetPackageOptions,
+	otherDeps: string[] = []
+): Promise<T | undefined> {
+	let packageImport;
+	try {
+		packageImport = await import(packageName);
+	} catch (e) {
+		info(
+			logging,
+			'',
+			`To continue, Astro requires the following dependency to be installed: ${bold(packageName)}.`
+		);
+		const result = await installPackage([packageName, ...otherDeps], options, logging);
+
+		if (result) {
+			packageImport = await import(packageName);
```

Comment:
I think nodejs caches the result if it import fails the first time, so this second import of `packageName` would not work. I think we can do an initial `require.resolve` first to check the existence, then install to workaround it

Reply:
Oh? It worked in my tests, hmm

Reply:
Here's the related issue: https://github.com/nodejs/modules/issues/307. I'm pretty sure it would hit that unless you already have `@astrojs/check` installed somewhere 🤔